tutorials/getting_started/semantic_parsing.md

```diff
@@ -0,0 +1,315 @@
+# The AllenNLP Semantic Parsing Framework
+
+_Semantic parsing_ is the task of mapping language to some kind of formal meaning representation.
+This meaning representation could be a logical statement in lambda calculus or lambda-DCS, a set of
+instructions for a robot to follow, or even a python, java, or SQL program.  In some cases these
+meaning representations are directly executable in some environment (like executing a query against
+a database, or running python code), and in others they are simply an attempt to normalize the
+semantics of a natural language utterance (like Abstract Meaning Representations or open-domain CCG
+semantics).  The thing all of these variations have in common is that they try to capture the
+meaning of language in some form, and there typically isn't a direct mapping from words in the
+utterance to the pieces of the meaning representation.  (TODO: add links above)
+
+We approach this problem in AllenNLP with encoder-decoder models, similar to the seq2seq models
+used in neural machine translation or summarization.  That is, we encode the input utterance (and
+whatever other context is available) using some neural encoder, then use a decoder to produce a
+statement in the output language. Because the output language in semantic parsing is a formal
+language, however, we use _constrained decoding_, forcing the model to only produce valid meaning
+representations.
```

Comment:
Could you give a small example of such a constraint here perhaps?

Reply:
I agree that having an example would make this a lot clearer, maybe a running example that's referred back to throughout. It could be referred to when talking about ``GrammarStatelet``, etc.